docs/style-spec/expressions.md

```diff
@@ -0,0 +1,40 @@
+# Style Function Expressions
+
+**Constants:**
+- LN2
+- PI
+- e
+
+**Literal:**
+- JSON string / number / boolean literal
+
+**Property lookup:**
+- Feature property:  `{ "ref": "feature", "key": propertyName, "default": defaultValue}`
```

Comment:
Syntax alternatives:

* `["data", key, default]`, where _key_ and _default_ are both expressions
* `["data", key]`, where _key_ is an expression, and we also provide a `["has", key]` expression, with which arbitrary default logic can be implemented.